cmd/frontend/internal/httpapi/lsif.go

```diff
@@ -1,26 +1,157 @@
 package httpapi
 
 import (
+	"crypto/hmac"
+	"crypto/sha256"
+	"strings"
+
+	"encoding/hex"
+	"encoding/json"
+
+	"fmt"
+
 	"net/http"
 	"net/http/httputil"
+	"net/url"
 
 	"github.com/gorilla/mux"
 	"github.com/sourcegraph/sourcegraph/cmd/frontend/backend"
+	"github.com/sourcegraph/sourcegraph/pkg/actor"
+	"github.com/sourcegraph/sourcegraph/pkg/api"
+	"github.com/sourcegraph/sourcegraph/pkg/env"
+	"github.com/sourcegraph/sourcegraph/pkg/extsvc/github"
+
+	"strconv"
 )
 
+var verificationGitHubToken = env.Get("LSIF_VERIFICATION_GITHUB_TOKEN", "", "The GitHub token that is used to verify that a user owns a repository.")
+var lsifUploadSecret = env.Get("LSIF_UPLOAD_SECRET", "", "Used to generate LSIF upload tokens. Must be long (160+ bits) to make offline brute-force attacks difficult.")
+var HasLSIFUploadSecret = lsifUploadSecret != ""
+
 func lsifProxyHandler(p *httputil.ReverseProxy) func(http.ResponseWriter, *http.Request) {
 	return func(w http.ResponseWriter, r *http.Request) {
 		r.URL.Path = mux.Vars(r)["rest"]
 		p.ServeHTTP(w, r)
 	}
 }
 
+func generateUploadToken(repoID api.RepoID) ([]byte, error) {
+	mac := hmac.New(sha256.New, []byte(lsifUploadSecret))
+	_, err := mac.Write([]byte(strconv.Itoa(int(repoID))))
```

Comment:
nit:

```suggestion
	_, err := mac.Write([]byte(strconv.FormatInt(int64(repoID), 10)))
```

This makes it so that if `api.RepoID` ever changes from 32-bit to 64-bit there would not be a subtle error here uncaught by type-checking.